File: 2018/02/2_2.rb
lines = File.read("input.txt").split("\n")
lines.combination(2) { |a, b|
    same = ""
    diffs = 0
    for i in 0..(a.length-1) do
        if a[i] != b[i] 
            diffs += 1
            break if diffs > 1
        else
            same += a[i]
        end
    end

    if diffs == 1
        puts same
        return
    end
}

File: 2021/02/2_2.py
lines = open('input.txt', 'r').read().splitlines()

aim = 0
x = 0
y = 0

for line in lines:
    tokens = line.split(' ')
    cmd = tokens[0]
    num = int(tokens[1])
    if cmd == "forward":
        x += num
        y += aim * num
    elif cmd == "up":
        aim -= num
    elif cmd == "down":
        aim += num
print(x*y)